pkg/ctl/gitops/apply.go

```diff
@@ -0,0 +1,178 @@
+	}
+
+	quickstartRepoURL, err := repoURLForQuickstart(opts.quickstartNameArg)
+	if err != nil {
+		return errors.Wrapf(err, "please supply a valid Quick Start name or url")
+	}
+
+	if err := cmdutils.NewGitopsMetadataLoader(cmd).Load(); err != nil {
+		return err
+	}
+	cfg := cmd.ClusterConfig
+	ctl, err := cmd.NewCtl()
+	if err != nil {
+		return err
+	}
+
+	if err := ctl.CheckAuth(); err != nil {
+		return err
+	}
+	if err := ctl.RefreshClusterConfig(cfg); err != nil {
+		return err
+	}
+	kubernetesClientConfigs, err := ctl.NewClient(cfg)
+	if err != nil {
+		return err
+	}
+	k8sConfig := kubernetesClientConfigs.Config
+
+	k8sRestConfig, err := clientcmd.NewDefaultClientConfig(*k8sConfig, &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return errors.Wrap(err, "cannot create Kubernetes client configuration")
+	}
+	k8sClientSet, err := kubeclient.NewForConfig(k8sRestConfig)
+	if err != nil {
+		return errors.Errorf("cannot create Kubernetes client set: %s", err)
+	}
+
+	// Create the flux installer. It will clone the user's repository in the outputPath
+	fluxOpts := flux.InstallOpts{
+		GitURL:      opts.git.URL,
+		GitBranch:   opts.git.Branch,
+		OutputPath:  opts.outputPath,
+		GitEmail:    opts.git.Email,
+		GitUser:     opts.git.User,
+		Namespace:   "flux",
+		GitFluxPath: "flux/",
+		Timeout:     defaultGitTimeout,
+	}
+	fluxInstaller := flux.NewInstaller(context.Background(), k8sRestConfig, k8sClientSet, &fluxOpts)
+
+	processor := &fileprocessor.GoTemplateProcessor{
+		Params: fileprocessor.NewTemplateParameters(cmd.ClusterConfig),
+	}
+
+	// Create the profile generator. It will output the processed templates into a new "/base" directory into the user's repo
+	usersRepoName, err := git.RepoName(opts.git.URL)
+	if err != nil {
+		return err
+	}
+	usersRepoDir := filepath.Join(opts.outputPath, usersRepoName)
+	profileOutputPath := filepath.Join(usersRepoDir, "base")
+
+	profile := &gitops.Profile{
+		Processor: processor,
+		Path:      profileOutputPath,
+		GitOpts: git.Options{
+			URL:    quickstartRepoURL,
+			Branch: "master",
+		},
+		GitCloner: git.NewGitClient(context.Background(), defaultGitTimeout),
+		FS:        afero.NewOsFs(),
+		IO:        afero.Afero{Fs: afero.NewOsFs()},
+	}
+
+	// A git client that operates in the user's repo
+	gitClient := git.NewGitClientDir(context.Background(), defaultGitTimeout, usersRepoDir)
+
+	gitOps := gitops.GitOps{
+		UsersRepoOpts:    opts.git,
+		GitClient:        gitClient,
+		ProfileGenerator: profile,
+		FluxInstaller:    fluxInstaller,
+		ClusterConfig:    cmd.ClusterConfig,
+		QuickstartName:   opts.quickstartNameArg,
+	}
+
+	if err = gitOps.Apply(context.Background()); err != nil {
+		return err
+	}
+	return nil
+}
+
+func repoURLForQuickstart(quickstartArgument string) (string, error) {
+	// FIXME: urls like git@github.com:weaveworks/eksctl produce the error "first path segment in URL cannot contain colon"
```

Comment:
What's the plan regarding this `FIXME`? Is it to fix it in this PR or in a follow-up issue? If the latter, it may be worth adding a link to the issue in the comment.

Reply:
I fixed it by using a small library that can parse git urls